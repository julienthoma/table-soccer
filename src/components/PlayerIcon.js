/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { GREY_DARK, GREY_LIGHT } from './Colors';

const HEAD_SIZE = 8;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1px;
`;

const Head = styled.div`
  background: ${props => (props.dark ? GREY_DARK : GREY_LIGHT)};
  height: ${HEAD_SIZE}px;
  width: ${HEAD_SIZE}px;
  border-radius: 50px;
`;

const Body = styled.div`
  background: ${props => (props.dark ? GREY_DARK : GREY_LIGHT)};
  margin-top: 1px;
  width: ${1.3 * HEAD_SIZE}px;
  height: ${1.8 * HEAD_SIZE}px;
  border-top-left-radius: ${HEAD_SIZE}px;
  border-top-right-radius: ${HEAD_SIZE}px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
`;

const PlayerIcon = ({ dark }) => (
  <Container>
    <Head />
    <Body />
  </Container>
);

PlayerIcon.propTypes = {
  dark: PropTypes.bool.isRequired
};

export default PlayerIcon;
