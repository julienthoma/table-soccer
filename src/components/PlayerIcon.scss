$headSize: 8px;
$color: #BDBDBD;

.root {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1px;
}

.head {
  background: $color;
  height: $headSize;
  width: $headSize;
  border-radius: 50px;
}

.body {
  margin-top: 1px;
  width: 1.3 * $headSize;
  height: 1.8 * $headSize;
  background: $color;
  border-top-left-radius: $headSize;
  border-top-right-radius: $headSize;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
}
