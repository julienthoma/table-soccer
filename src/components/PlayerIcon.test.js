import React from 'react';
import { renderWithMuiTheme } from '../test.utils';
import PlayerIcon from './PlayerIcon';

it('PlayerIcon renders correctly', () => {
  expect(renderWithMuiTheme(<PlayerIcon dark />)).toMatchSnapshot();
});
