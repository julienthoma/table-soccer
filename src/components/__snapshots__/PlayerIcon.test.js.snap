// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerIcon renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1px;
}

.c1 {
  background: #c5c5c5;
  height: 8px;
  width: 8px;
  border-radius: 50px;
}

.c2 {
  background: #c5c5c5;
  margin-top: 1px;
  width: 10.4px;
  height: 14.4px;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;
